### ISSUE TITLE
Cannot get NPC to sync

### ISSUE THREAD

**Issue by bkeys** [godotengine/godot#79302]
### Godot version

4.1

### System information

Fedora 6.2.9-300.fc38.ppc64le AMD GPU

### Issue description

I am unable to sync the position of my zombies. All examples out there only spawn players but none spawn NPCs. I am trying to have the NPCs be under the authority of the server (since that is the only thing that makes sense), but regardless I cannot get the position of the zombies to be sync'ed by a multiplayersynchronizer node. I get no real errors except this message, which looking through the C++ I cannot find out what I am doing wrong:
```
E 0:00:05:0399   on_spawn_receive: Condition "parent->has_node(name)" is true. Returning: ERR_INVALID_DATA
  <C++ Source>   modules/multiplayer/scene_replication_interface.cpp:596 @ on_spawn_receive()
```

### Steps to reproduce

Just run the project, shoot the zombie and you will see that they report their positions and they are not the same at all on the server as opposed to the client

### Minimal reproduction project

[Zombie2.zip](https://github.com/godotengine/godot/files/12004658/Zombie2.zip)

**Comment by TheYellowArchitect** [comment]
Hello!

I tested the above MRP. I suggest having both host+join buttons, because --server at args isn't as simple :P
Anyway, I couldn't get to your issue because of this
![non-existent-rpc-function](https://github.com/godotengine/godot/assets/25630803/bb1c7a39-1a40-4db3-a845-3f581cf6fd51)

I haven't had a real use-case myself for syncing movement (I am doing turn-based and level editors online atm), but if indeed it doesnt sync properly, perhaps its because the server is always some frames ahead of clients? In localhost it shouldn't be more than 1~2 but anyway, if you fix the above MRP, do send so I give it a try again. I am curious myself if the above is enough, or a snapshot system is required like most multiplayer games do

**Comment by TheYellowArchitect** [comment]
Also for certainty, add 4 arguments to the RPC annotations instead of 2
```gdscript
func recieve_damage():
	health -= 1
	print(name + " hit")
	if health <= 0:
		print(name + " I am a dead man")
		queue_free()
	health_changed.emit(health)
	print(position)


@rpc("any_peer", "call_remote")
func set_pasdfasdf(position_):
	position = position_
```

**Comment by bkeys** [comment]
The code you are showing was me trying to rig something to work for the time being. I am trying to sync the positions using a MultiplayerSynchronizer not RPCs. Here is an updated MRP
[Zombie2.zip](https://github.com/godotengine/godot/files/12005649/Zombie2.zip)
I'd think the frame the server is on shouldn't matter; because not all players are joining at the same time, someone should be able to drop it and get everything spawned and sync'ed upon joining

**Comment by TheYellowArchitect** [comment]
I tweaked the button so I test fast via host/join combo. Basically the host button calls this code:
```gdscript
func _init_server():
	main_menu.hide()
	hud.show()
	enet_peer.create_server(PORT)
	multiplayer.multiplayer_peer = enet_peer
	multiplayer.peer_connected.connect(add_player)
	multiplayer.peer_disconnected.connect(remove_player)
	
	print("server init'ed")
	$ZombieSpawnTimer.autostart = true
	$ZombieSpawnTimer.start()
	```
I screen recorded my testing, and I couldn't get the zombie to even spawn on client. I hope at least this video is useful for debugging feedback or sth :(

**Comment by Jordyfel** [comment]
The issue seems to mean that there is already a node with such a name when the replication packet arrives. Your project is not a minimum reproduction project at all, it doesn't even allow starting a multiplayer game without command line args.

What is this statement supposed to do: `set_multiplayer_authority(str(name).to_int())`

**Comment by bkeys** [comment]
> The error message seems to mean that there is already a node with such a name when the replication packet arrives. Your project is not a minimum reproduction project at all, it doesn't even allow starting a multiplayer game without command line args.
> 
> What is this statement supposed to do: `set_multiplayer_authority(str(name).to_int())`

Sorry about the crappy MRP. I don't know what that line is supposed to do, it came from a tutorial that I followed.
As for the nodes name being the same, isn't that how it's supposed to be since it's replicating a scene? Are you sure that is what the error is saying?

@TheYellowArchitect here is another project, the zombie spawns in this one, and replicates the original issue. I apologize for the runaround. I'd be very grateful if you (or anyone) took a look, the documentation on multiplayer isn't the most plentiful and I'm doing my best to understand this. With the functionality being so new it's hard to decipher what is my being new at it and what is a bug.
[Zombie2.zip](https://github.com/godotengine/godot/files/12022774/Zombie2.zip)

**Comment by TheYellowArchitect** [comment]
I understand. I am curious if this bug really exists. I will try out the MRP in the upcoming days

**Comment by bkeys** [comment]
> The error message seems to mean that there is already a node with such a name when the replication packet arrives. Your project is not a minimum reproduction project at all, it doesn't even allow starting a multiplayer game without command line args.
> 
> What is this statement supposed to do: `set_multiplayer_authority(str(name).to_int())`

Sorry about the crappy MRP. I don't know what that line is supposed to do, it came from a tutorial that I followed.
As for the nodes name being the same, isn't that how it's supposed to be since it's replicating a scene? Are you sure that is what the error is saying?

@TheYellowArchitect here is another project, the zombie spawns in this one, and replicates the original issue. I apologize for the runaround. I'd be very grateful if you (or anyone) took a look, the documentation on multiplayer isn't the most plentiful and I'm doing my best to understand this. With the functionality being so new it's hard to decipher what is my being new at it and what is a bug.

[Zombie2.zip](https://github.com/godotengine/godot/files/12040252/Zombie2.zip)

**Comment by TheYellowArchitect** [comment]
The MRP is still incomplete because you can only host via command args, and this line throws error:
`Input.mouse_mode = Input.MOUSE_MODE_CAPTURED`
I commented it out, but mayhaps its an unreported bug, anyway. I upload the MRP with the above 2 fixes (and addition of a debug message) for anyone wishing to test: 
[proper-MRP.zip](https://github.com/godotengine/godot/files/12060899/proper-MRP.zip)

As for the bug, it's confirmed to not be a bug. You have not used MultiplayerSynchronizer (see MultiplayerSpawner), nor do you sync movement via RPCs. So, your position is not synced at all across the network, each instance has its own position. It is expected to have different positions at the same frame since the physics of the host and client are not deterministic and not at the same timeframe.

This is your zombie's position logic:
```gdscript
func _physics_process(delta):
	# Add the gravity.
	if not is_on_floor():
		velocity.y -= gravity * delta
	var direction = Vector3()
	var npos = position
	npos.x -= 10
	npos.z -= 10
	nav.target_position = npos
	direction = nav.get_next_path_position() - global_position
	direction = direction.normalized()
	velocity = velocity.lerp(direction * speed, accel * delta)
	move_and_slide()
	#print($MultiplayerSynchronizer.get_multiplayer_authority())
```

Nowhere does it sync. The positions are independent. The same applies to your player/avatar, if more players join, they are not synced.

tl;dr: Use Multiplayer Synchronizer node, or use unreliable RPCs to synchronize position.

@Sauermann @Calinou This issue was tested and it's not a bug, so you can close it
